src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json

```diff
@@ -0,0 +1,557 @@
+                        }
+                    }
+                   ]
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                       {
+                        "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('name_nic'))]"
+                    }
+                   ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[reference(resourceId('Microsoft.Storage/storageAccounts/', variables('name_storageAccount')), '2019-06-01').primaryEndpoints.blob]"
+                    }
+                }
+            },
+            "plan": {
+                "name": "[variables('name_linuxImageOfferSKU')]",
+                "publisher": "[variables('const_imagePublisher')]",
+                "product": "[variables('const_imageOffer')]"
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "name": "[concat(parameters('adminVMName'),'/newuserscript')]",
+            "apiVersion": "2019-07-01",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[concat('Microsoft.Compute/virtualMachines/', parameters('adminVMName'))]"
+            ],
+            "properties": {
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.0",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                    "fileUris": [
+                        "[uri(parameters('_artifactsLocationAdminTemplate'), concat('../scripts/', variables('name_scriptFile'), parameters('_artifactsLocationSasToken')))]"
+                    ],
+                    "commandToExecute": "[concat('sh',' ',variables('name_scriptFile'),' ',parameters('wlsDomainName'),' ',parameters('wlsUserName'),' ',parameters('wlsPassword'),' ',parameters('adminVMName'),' ', variables('const_wlsHome'),' ',string(parameters('enableHTTPAdminListenPort')),' ',string(parameters('enableCustomSSL')),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomIdentityKeyStoreData')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomIdentityKeyStorePassPhrase')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomIdentityKeyStoreType')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomTrustStoreData')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomTrustStorePassPhrase')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomTrustStoreType')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLServerPrivateKeyAlias')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLServerPrivateKeyPassPhrase')),''),' ')]"
+                }
```

Comment:
See https://github.com/wls-eng/arm-oraclelinux-wls/issues/158 Move `commandToExecute` properties that include passwords to `protectedSettings`

Reply:
Thanks Ed. I will move the commandToExecute properties under protectedSettings. 

Regarding changes to createUIDefinition.json, as you have mentioned, we would require additional fields that will accept user inputs regarding keyvault resource group and keyvault data. 

The template though works without them as well, as I have made them optional and also defaulted the customSSLEnabled flag to false.

I will retest the changes for protected settings and create a new pull request. You may want to reject the existing merge request.